v6_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,226 @@
+
+## DropInRequest
+
+The getters and setters on `DropInRequest` have been renamed with a consistent get/set pattern to allow for Kotlin synthesized properties. 
+See the examples below for a full list of optional parameters:
+
+Java:
+```java
+    DropInRequest dropInRequest = new DropInRequest();
+    dropInRequest.setShouldCollectDeviceData(true);
+    dropInRequest.setGooglePayRequest(googlePayRequest);
+    dropInRequest.setGooglePayDisabled(true);
+    dropInRequest.setPayPalRequest(paypalRequest);
+    dropInRequest.setPayPalDisabled(true);
+    dropInRequest.setVenmoDisabled(true);
+    dropInRequest.setCardDisabled(true);
+    dropInRequest.setShouldRequestThreeDSecureVerification(true);
+    dropInRequest.setThreeDSecureRequest(threeDSecureRequest);
+    dropInRequest.setShouldMaskCardNumber(true);
+    dropInRequest.setShouldMaskSecurityCode(true);
+    dropInRequest.setEnableVaultManager(true);
+    dropInRequest.setAllowVaultCardOverride(true);
+    dropInRequest.setVaultCardDefaultValue(true);
+    dropInRequest.setCardholderNameStatus(CardForm.FIELD_OPTIONAL);
+    dropInRequest.setVaultVenmoDefaultValue(true);
+```
+
+Kotlin:
+```kotlin
+    val dropInRequest = DropInRequest()
+    dropInRequest.shouldCollectDeviceData = true
+    dropInRequest.googlePayRequest = googlePayRequest
+    dropInRequest.googlePayDisabled = true
+    dropInRequest.payPalRequest = paypalRequest
+    dropInRequest.payPalDisabled = true
+    dropInRequest.venmoDisabled = true
+    dropInRequest.cardDisabled = true
+    dropInRequest.shouldRequestThreeDSecureVerification = true
+    dropInRequest.threeDSecureRequest = threeDSecureRequest
+    dropInRequest.shouldMaskCardNumber = true
+    dropInRequest.shouldMaskSecurityCode = true
+    dropInRequest.enableVaultManager = true
+    dropInRequest.allowVaultCardOverride = true
+    dropInRequest.vaultCardDefaultValue = true
+    dropInRequest.cardholderNameStatus = FIELD_OPTIONAL
+    dropInRequest.vaultVenmoDefaultValue = true
+```
+
+The full list of changed parameters is below:
+1. `clientTokent` -> removed
+1. `tokenizationKey` -> removed
+1. `amount` -> removed
+1. `intent` -> removed
+1. `collectDeviceData` -> `shouldCollectDeviceData`
+1. `googlePaymentRequest` -> `googlePayRequest`
+1. `disableGooglePayment` -> `googlePayDisabled`
+1. `paypalRequest` -> `payPalRequest`
+1. `disablePayPal` -> `payPalDisabled`
+1. `disableVenmo` -> `venmoDisabled`
+1. `disableCard` -> `cardDisabled`
+1. `requestThreeDSecureVerification` -> `shouldRequestThreeDSecureVerification`
+1. `maskCardNumber` -> `shouldMaskCardNumber`
+1. `maskSecurityCode` -> `shouldMaskSecurityCode`
+1. `vaultManager` -> `enableVaultManager`
+1. `vaultCard` -> `vaultCardDefaultValue`
+1. `vaultVenmo` -> `vaultVenmoDefaultValue`
+
+## Authorization
+
+The `clientToken`, `tokenizationKey`, and `authorization` fields have been removed from `DropInRequest`. 
+In v6, authorization should be included when instantiating a `DropInClient` instead.
```

Comment:
Can we include a code snippet here?

Reply:
Also, maybe this section should go below the section where we introduce `DropInClient`. Also, might be helpful if `DropInClient` has it's own section. We could have a code snippet comparing the old to new integration pattern so they can be clear how they should migrate away from `BraintreeFragment`.

Reply:
I added a small section for `DropInClient` without any code snippets since they would be the same snippets from the authorization and the launch drop in sections. I think it makes sense now that a reader learns about the client, sees how to instantiate with authorization, and then sees how to launch Drop-in. `BraintreeFragment` wasn't a merchant-facing part of Drop-in, so I don't think it makes sense to put that in the migration guide here, but I added a note about no longer needing an `Intent` to launch Drop-in. Let me know if this looks better.

Reply:
Ah gotcha - thanks for clarifying!